test/sha1_test.dart

```diff
@@ -2,50 +2,12 @@
 // for details. All rights reserved. Use of this source code is governed by a
 // BSD-style license that can be found in the LICENSE file.
 
-import 'dart:async';
-import 'dart:convert';
-
 import 'package:crypto/crypto.dart';
-import 'package:test/test.dart';
-
-void main() {
-  group('with a chunked converter', () {
-    test('add may not be called after close', () {
-      var sink =
-          sha1.startChunkedConversion(new StreamController<Digest>().sink);
-      sink.close();
-      expect(() => sink.add([0]), throwsStateError);
-    });
 
-    test('close may be called multiple times', () {
-      var sink =
-          sha1.startChunkedConversion(new StreamController<Digest>().sink);
-      sink.close();
-      sink.close();
-      sink.close();
-      sink.close();
-    });
+import 'test_utils.dart';
 
-    test('close closes the underlying sink', () {
-      var inner = new ChunkedConversionSink<Digest>.withCallback(
-          expectAsync1((accumulated) {
-        expect(accumulated.length, equals(1));
-        expect(accumulated.first.toString(),
-            equals('da39a3ee5e6b4b0d3255bfef95601890afd80709'));
-      }));
-
-      var outer = sha1.startChunkedConversion(inner);
-      outer.close();
-    });
-  });
-
-  group('standard vector', () {
-    for (var i = 0; i < _inputs.length; i++) {
-      test(_digests[i], () {
-        expect(sha1.convert(_inputs[i]).toString(), equals(_digests[i]));
-      });
-    }
-  });
+void main() {
+  testHash(sha1, _inputs, _digests, 'da39a3ee5e6b4b0d3255bfef95601890afd80709');
```

Comment:
This is terser, but I don't think it's clearer. I don't think the DRY is worth the indirection here.